packages/synapse-react-client/src/components/EntityCitation/EntityCitation.test.tsx

```diff
@@ -0,0 +1,259 @@
+const mockUseGetDOIAssociation = vi.mocked(useGetDOIAssociation)
+
+const {
+  mock: useGetEntityBundleMockImpl,
+  setSuccess: setMockUseGetEntityBundleSuccess,
+} = getUseQueryMock<Partial<EntityBundle>, SynapseClientError>()
+
+const { mock: useGetEntityDOIMockImpl, setSuccess: setEntityDOISuccess } =
+  getUseQueryMock<DoiAssociation | null, SynapseClientError>()
+
+const { mock: useGetProjectDOIMockImpl, setSuccess: setProjectDOISuccess } =
+  getUseQueryMock<DoiAssociation | null, SynapseClientError>()
+
+const mockEntityWithUnversionedDoiId = 'syn61841662'
+const mockEntityWithVersionedDoiId = 'syn66268092'
+const mockEntityWithNoDoiId = 'syn68243871'
+const mockProjectWithNoDoiId = 'syn68243871'
+const mockProjectWithDoiId = 'syn68244561'
+
+const doiEntitySuccess: DoiAssociation = {
+  objectId: mockEntityWithUnversionedDoiId,
+  objectType: DoiObjectType.ENTITY as ObjectType,
+  doiUri: '10.7303/syn61841662',
+  associationId: '',
+  etag: '',
+  doiUrl: '',
+  associatedBy: '',
+  associatedOn: '',
+  updatedBy: '',
+  updatedOn: '',
+}
+
+const fileWithDoiAssociation: Partial<EntityBundle> = {
+  entity: {
+    id: 'syn61841662',
+    name: 'blackcat.jpg',
+    concreteType: 'org.sagebionetworks.repo.model.FileEntity',
+  },
+  doiAssociation: doiEntitySuccess,
+  entityType: EntityType.FILE,
+}
+
+const versionedDoiEntitySuccess: DoiAssociation = {
+  objectId: mockEntityWithVersionedDoiId,
+  objectType: DoiObjectType.ENTITY as ObjectType,
+  doiUri: '10.7303/syn66268092.1',
+  associationId: '',
+  etag: '',
+  doiUrl: '',
+  associatedBy: '',
+  associatedOn: '',
+  updatedBy: '',
+  updatedOn: '',
+}
+
+const doiProjectSuccess: DoiAssociation = {
+  objectId: 'syn64042437',
+  objectType: DoiObjectType.ENTITY as ObjectType,
+  doiUri: '10.7303/syn64042437',
+  associationId: '',
+  etag: '',
+  doiUrl: '',
+  associatedBy: '',
+  associatedOn: '',
+  updatedBy: '',
+  updatedOn: '',
+}
+
+const openPopover = async (buttonName: string) => {
+  const button = screen.getByRole('button', { name: `${buttonName}` })
+  await userEvent.click(button)
+}
+
+describe('EntityCitation tests', () => {
+  beforeEach(() => {
+    vi.clearAllMocks()
+
+    mockUseGetEntityBundle.mockImplementation(useGetEntityBundleMockImpl)
+
+    mockUseGetDOIAssociation.mockImplementation((request, options) => {
+      if (
+        request.id === mockEntityWithVersionedDoiId &&
+        request.version === 1
+      ) {
+        return useGetEntityDOIMockImpl()
+      } else if (
+        request.id === mockEntityWithUnversionedDoiId &&
+        request.version === undefined
+      ) {
+        return useGetEntityDOIMockImpl()
+      } else if (
+        request.id === mockProjectWithDoiId &&
+        request.version === undefined
+      ) {
+        return useGetProjectDOIMockImpl()
+      } else if (
+        request.id === mockProjectWithNoDoiId &&
+        request.version === undefined
+      ) {
+        return useGetProjectDOIMockImpl()
+      }
```

Comment:
Since you're setting the mock response body in each test, isn't this the same thing, but more simple?
```suggestion
      if (request.id === mockEntityWithVersionedDoiId) {
        return useGetEntityDOIMockImpl()
      } else if (request.id === mockProjectWithDoiId) {
        return useGetProjectDOIMockImpl()
      }
```